code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm

```diff
@@ -158,6 +159,8 @@
 			new /obj/effect/mob_spawn/corpse/human/charredskeleton(T)
 		else if(dwarf_mob)
 			new /obj/effect/mob_spawn/corpse/human/legioninfested/dwarf(T)
+		else if(snow_legion)
+			new /obj/effect/mob_spawn/corpse/human/legioninfested/snow(T)
 		else
 			new /obj/effect/mob_spawn/corpse/human/legioninfested(T)
```

Comment:
On reflection we should definitely refactor this to work another way, but I may just put that on my todo list rather than making you do it given the rigamarole this has already been for you.